governance/src/agent.js

```diff
@@ -0,0 +1,301 @@
+const { Finding, ethers } = require('forta-agent');
+
+const agentConfig = require('../agent-config.json');
+
+const { getAbi, createProposalFromLog } = require('./utils');
+
+// set up a variable to hold initialization data used in the handler
+const initializeData = {};
+
+// alert for when a new governance proposal is created
+function proposalCreatedFinding(proposal, address, config) {
+  return Finding.fromObject({
+    name: `${config.protocolName} Governance Proposal Created`,
+    description: `Governance Proposal ${proposal.proposalId} was just created`,
+    alertId: `${config.developerAbbreviation}-${config.protocolAbbreviation}-PROPOSAL-CREATED`,
+    type: 'Info',
+    severity: 'Info',
+    protocol: config.protocolName,
+    metadata: {
+      address,
+      ...proposal,
+    },
+  });
+}
+
+function voteCastFinding(voteInfo, address, config) {
+  let description = `Vote cast with weight ${voteInfo.weight.toString()}`;
+  switch (voteInfo.support) {
+    case 0:
+      description += ' against';
+      break;
+    case 1:
+      description += ' in support of';
+      break;
+    case 2:
+      description += ' abstaining from';
+      break;
+    default:
+      description += ` with unknown support "${voteInfo.support}" for`;
+  }
+  description += ` proposal ${voteInfo.proposalId}`;
+
+  return Finding.fromObject({
+    name: `${config.protocolName} Governance Proposal Vote Cast`,
+    description,
+    alertId: `${config.developerAbbreviation}-${config.protocolAbbreviation}-VOTE-CAST`,
+    type: 'Info',
+    severity: 'Info',
+    protocol: config.protocolName,
+    metadata: {
+      address,
+      voter: voteInfo.voter,
+      weight: voteInfo.weight.toString(),
+      reason: voteInfo.reason,
+    },
+  });
+}
+
+function proposalCanceledFinding(proposalId, address, config) {
+  return Finding.fromObject({
+    name: `${config.protocolName} Governance Proposal Canceled`,
+    description: `Governance proposal ${proposalId} has been canceled`,
+    alertId: `${config.developerAbbreviation}-${config.protocolAbbreviation}-GOVERNANCE-PROPOSAL-CANCELED`,
+    type: 'Info',
+    severity: 'Info',
+    protocol: config.protocolName,
+    metadata: {
+      address,
+    },
+  });
+}
+
+function proposalExecutedFinding(proposalId, address, config) {
+  return Finding.fromObject({
+    name: `${config.protocolName} Governance Proposal Executed`,
+    description: `Governance proposal ${proposalId} has been executed`,
+    alertId: `${config.developerAbbreviation}-${config.protocolAbbreviation}-GOVERNANCE-PROPOSAL-EXECUTED`,
+    type: 'Info',
+    severity: 'Info',
+    protocol: config.protocolName,
+    metadata: {
+      address,
```

Comment:
Same comment as the one in `proposalCanceledFinding`

Reply:
Done.